### PR TITLE
WIP: Dino ip adapter training

### PR DESCRIPTION
**Pull request by isamu-isozaki**
Still very much WIP. Much code taken from @Laurent2916's pr. The tasks left are

- [x] validation 
- [x] checkpoint loading for just image encoder(dinov2) for IP adapter
- [x] Adding timestep+pooled text emb idea from @limiteinductive 
- [x] Setup toml to be the dinov2 ver and not color palette
- [x] Load initial checkpoint from IP adapter and fine-tune from that(handling shape diff etc)
- [x] Log gradient norms/tensor norms to wandb
- [x] Test out with small dinov2 model locally

For features, I added minsnr and input perturbation/embed dropping idea given from the original IP adapter repo so far since I heard from the laion discord that they are pretty useful. I also might add

- [x] Pre-encoding embeds
- [ ] Webdataset for around 40% increase in data query speed
- [ ] timestep bias

### CODE REVIEW

**Comment by isamu-isozaki** [comment]
One thing I should check is if requires_grad will work properly in my current design for the adapter.

**Comment by isamu-isozaki** [comment]
ok! The main features done. I'll start testing around tomorrow morning

**Comment by Laurent2916** [comment]
Work continues at https://github.com/isamu-isozaki/dino-ip-adapter-refiners.
Bounty is still active.
